### DISCUSSION TITLE
Should datetime value be undefined when time value is deleted?

### DISCUSSION THREAD

**Discussion by vytautas-pilk** [adobe/react-spectrum#3876] [Q&A] [answered]
When you provide any valid date and time value to `<DatePicker />`, for example `parseDateTime('2022-10-10T09:00:00')`, it stores that value in its state as `CalendarDate` type. Now, when you delete any of time values in `<DateField />` - hours or minutes, state value becomes ~~`undefined`~~ `null`*, even though date value (yyyy-MM-dd) is still rendered just fine. Is that normal behaviour?

It kind of messes up any form where dates are involved. You can set up validations, of course, but from a UI perspective, IMO, it doesn't make any sense at all.

\* edit

**Comment by LFDanLu** [comment, marked as answer]
IMO this is normal behavior since your DatePicker's date value is no longer fully defined once you delete the time values and it feels important to distinguish that value state rather than substitute `0:00:00`. To clarify, is the DatePicker you have in your example case use `granularity="minute"`?

**Reply by vytautas-pilk** [reply]
Hi, thanks for chiming in.

`granularity` is set to `second`, but does that make a difference?
What I find odd is that a remnant of previously valid date or time is still rendered, although `datePickerState` says it's `null`. Deleted date values become literally `yyyy`, `MM` or `dd` (in my case, because I'm working with ISO 8601 dates) and deleted time values become `--` between `:`s.

**Comment by vytautas-pilk** [comment]
@LFDanLu, after giving it a n-th though, I switched sides.

Purely because clearing every single date/time field after deleting a value is counter-productive if you'd like to change a tiny part of it.

Thank you again.
